### ISSUE TITLE
X-XSRF-TOKEN as well as API issues

### ISSUE THREAD

**Issue by jonfinley** [theotherp/nzbhydra2#54]
So everything works fine just using NZbhydra using ip:5076

But when I try to access using my reverse proxy domain.com/nzbhydra2 I get a 
`Access denied to IP [WAN]: Invalid CSRF Token 'null' was found on the request parameter '_csrf' or header 'X-XSRF-TOKEN'.`

I also cannot seem to connect sonarr/radarr using http://127.0.0.1:5076/nzbhydra

**Comment by jonfinley** [comment]
API notification: `HTTP Error - Res: [GET] http://127.0.0.1:5076/nzbhydra/api?t=movie&cat=2000,2010,2020,2030,2035,2040,2045,2050,2060&extended=1&apikey=(removed)&offset=0&limit=100: 404.NotFound`

**Comment by theotherp** [comment]
Did you configure it as described in the wiki?

Am 11.01.2018 3:24 vorm. schrieb "jonfinley" <notifications@github.com>:

**Comment by jonfinley** [comment]
yep!! So I also been reading some of the other issues, could this be related to the migration???

**Comment by theotherp** [comment]
No.

I'll need to do some tests.

Please create a new issue for the API error and follow the issue template, especially the note regarding debug infos.

**Comment by theotherp** [comment]
I'll post some more instructions later on how to provide more information on this issue.

**Comment by theotherp** [comment]
Please do as follows:

In your browser (either Firefox or Chrome) open a new tab, press F12 to open the developer tools, go to the network tab. Enter your local address for NZBHydra. In the network tab it will show all HTTP requests. Select the first one, press right (or open the context menu however necessary), select "Copy -> Copy request headers". Paste the content of your clipboard. Do the same for "Copy -> Copy response headers".

Remove any sensitive information from the text, e.g. your domain.

Example:
![firefox copy request headers](https://user-images.githubusercontent.com/14368889/34809709-f11d6056-f696-11e7-9981-ba16166f851f.png)

**Comment by jonfinley** [comment]
:authority:DOMAIN.COM
:method:GET
:path:/nzbhydra2/
:scheme:https
accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
accept-encoding:gzip, deflate, br
accept-language:en-US,en;q=0.9
authorization:Basic ZmlubGV5YWRtaW46SWdhdmVoZXIyLUZpbmxleQ==
cookie:organizrLanguage=en; PHPSESSID=qeo01a5694gp5gd0ple0rllek7; organizrToken=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImp0aSI6IjRmMWcyM2ExMmFhIn0.eyJpc3MiOiJPcmdhbml6ciIsImF1ZCI6Ik9yZ2FuaXpyIiwianRpIjoiNGYxZzIzYTEyYWEiLCJpYXQiOjE1MTU2MDU5NTksImV4cCI6MTUxNTY5MjM1OSwidXNlcm5hbWUiOiJmaW5sZXlhZG1pbiIsImdyb3VwIjoiQWRtaW4iLCJncm91cElEIjowLCJlbWFpbCI6InNlcnZlckBmaW5sZXkuZXN0YXRlIiwiaW1hZ2UiOiJodHRwczpcL1wvd3d3LmdyYXZhdGFyLmNvbVwvYXZhdGFyXC8xZmU5ZmYxNDQzNjgzMmMxZWVmNGJkYjAwYjBjZWU4Nz9zPTEwMCZkPW1tIn0.2YgRleeqZu-6yIg0Km_ioOzQ0swebarHtkYiemyQ054; _pk_id.1.4f7a=15bd727c908928cd.1515635149.2.1515642534.1515638412.; remember-me=aUZpREk5V2diVVV1eWhpdUFRbFJyQT09Ok5EN0QwM2tXeTdVaW1VVjhIY2NYc0E9PQ; XSRF-TOKEN=f5296d25-0e2f-4adc-aa0f-0459fc503547; JSESSIONID=981FA7F888B3ED207EF2285366D711AB
upgrade-insecure-requests:1
user-agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36

I assume this is what you wanted: The option to copy request headers was not there.

**Comment by theotherp** [comment]
See other issue, you call the wrong path.

**Comment by jonfinley** [comment]
I just corrected that. it's still not working correctly.

**Comment by theotherp** [comment]
For some reason the XSRF token is different for those addresses. You could try deleting the cookies:

![img](https://i.imgur.com/LPLsSw1.png)

The next version should fix this issue.

**Comment by theotherp** [comment]
Turns out I'm half smart and added a switch to hopefully fix this but forgot to add it to the UI...

Shutdown Hydra, open your `nzbhydra.yml`, use for `useCsrf` and set it to false. Start Hydra again. See if it works now.

**Comment by jonfinley** [comment]
That worked there. Can now access via domain.com/nzbhydra2 without errors

**Comment by theotherp** [comment]
Glad to hear. I'll add it to the UI in the next version.

**Comment by 1activegeek** [comment]
So coming to pile onto this. I have likely a very similar setup to jon. Running nginx with Hydra2 as a RP entry. I was able to get Sonarr/Radarr connected, but adding the /hydra to the end of the URL. The problem I'm finding now though, is that when I try to connect to Hydra through my RP, I keep seeing a popup message as seen below. I'm also seeing lines similar below. I'm incapable of browsing to the history/stats tab either. That message just keeps popping up when I attempt to browse to it. 

I've adjusted the CSRF option as you indicated and it works. The odd thing is that I was able to remove the extra /hydra from my RP entry, but it is still required for Sonarr/Radarr. I guess I'm just confused why that extra base URL is needed in V2 and not in V1? But also, why the CSRF issue was only appearing on the history/stats page? 

```
[2m2018-03-20 22:41:23.266[0;39m [33m WARN[0;39m [2m---[0;39m [2m[0.0-5076-exec-1][0;39m [36mo.n.auth.AuthAndAccessEventHandler [0;39m [2m:[0;39m [35m[0;39mAccess denied to IP null: Invalid CSRF Token 'null' was found on the request parameter '_csrf' or header 'X-XSRF-TOKEN'.
[2m2018-03-20 22:41:23.267[0;39m [33m WARN[0;39m [2m---[0;39m [2m[0.0-5076-exec-1][0;39m [36mo.n.auth.LoginAndAccessAttemptService [0;39m [2m:[0;39m [35m[0;39mUnable to log failed login by empty IP
```

![screen shot 2018-03-20 at 10 45 40 pm](https://user-images.githubusercontent.com/6810720/37692662-7a14aad4-2c90-11e8-8a15-7b8b5cfa8678.png)
